tower-balance/src/lib.rs

```diff
@@ -312,6 +313,48 @@ impl<F: Future, E> Future for ResponseFuture<F, E> {
     }
 }
 
+
+// ===== impl Error =====
+
+impl<T, U> fmt::Display for Error<T, U>
+where
+    T: fmt::Debug,
+    U: fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Inner(ref why) =>
+                write!(f, "inner service error: {:?}", why),
```

Comment:
Should this not use `Display` as well? The inner errors should impl `Display` as they are also errors (as perf the impl below).

Reply:
Yeah, you're right, that's probably more correct. When I started on this branch, I wasn't sure how far down the stack I was going to have to add `Error` impls, so I wasn't sure if all the inner errors would be `Display`. Will change it.